app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -20,7 +20,7 @@ def initialize(hosts, template_invocation)
     # with all hosts.
     def to_hash
       hosts = @hosts.map do |h|
-        RemoteExecutionProvider.find_ip_or_hostname(h)
+        AnsibleProvider.find_ip_or_hostname(h)
```

Comment:
Does it make sense to call it `find_ip_or_hostname` if we never expect `ip` in here? I would suggest calling `find_fqdn` directly

Reply:
Calling `find_ip_or_hostname` will raise an exception in case the host doesn't have fqdn associated with any of its interfaces. Not sure if that can even happen. We might simplify this even further and just use `host.name` instead

Reply:
I would be ok with that